common/functions-v2/src/scalars/string.rs

```diff
@@ -156,3 +289,48 @@ fn vectorize_string_to_string(
         }
     }
 }
+
+// Vectorize (string, string) -> string function with customer estimate_bytes.
+// This function is only used for trim_[leading|trailing|both] function.
+// When arg1 is a scalar, arg2 must be a scalar.
+fn vectorize_two_string_to_string_from_trim(
```

Comment:
is it possible to make it more general like `vectorize_string_to_string_2_arg`? There is an example in `common_expression::functions` named `vectorize_2_arg`.

Reply:
Let me take a look.

Reply:
We can make it even further, since it's possible to make all args generic, and only the output have to be string type.

Reply:
I can make it more general, and remove the (arg1: Scalar, arg2: Scalar) cases in unit tests of trim functions.

Reply:
the case is possible in constant folding.